bitbots_blackboard/src/bitbots_blackboard/capsules/world_model_capsule.py

```diff
@@ -207,35 +208,30 @@ def get_ball_angle(self):
     def get_ball_speed(self):
         raise NotImplementedError
 
-    def balls_callback(self, msg: PoseWithCertaintyArray):
-        if msg.poses:
-            balls = sorted(msg.poses, reverse=True, key=lambda ball: ball.confidence)  # Sort all balls by confidence
-            ball = balls[0]  # Ball with highest confidence
-
-            if ball.confidence == 0:
-                return
-
-            ball_buffer = PointStamped(msg.header, ball.pose.pose.position)
-            try:
-                self.ball = self.tf_buffer.transform(ball_buffer, self.base_footprint_frame,
-                                                     timeout=rospy.Duration(0.3))
-                self.ball_odom = self.tf_buffer.transform(ball_buffer, self.odom_frame, timeout=rospy.Duration(0.3))
-                self.ball_map = self.tf_buffer.transform(ball_buffer, self.map_frame, timeout=rospy.Duration(0.3))
-                # Set timestamps to zero to get the newest transform when this is transformed later
-                self.ball_odom.header.stamp = rospy.Time(0)
-                self.ball_map.header.stamp = rospy.Time(0)
-                self.ball_seen_time = rospy.Time.now()
-                self.ball_seen = True
+    def ball_filtered_callback(self, msg: PoseWithCovarianceStamped):
+        self.ball_filtered = msg
 
-            except (tf2.ConnectivityException, tf2.LookupException, tf2.ExtrapolationException) as e:
-                rospy.logwarn(e)
+        # When the precision is not sufficient, the ball ages.
+        x_sdev = msg.pose.covariance[0]  # position 0,0 in a 6x6-matrix
+        y_sdev = msg.pose.covariance[7]  # position 1,1 in a 6x6-matrix
+        if x_sdev > self.body_config['ball_position_precision_threshold']['x_sdev'] or \
```

Comment:
The covariance gets checked for both the position and the twist. Both times the cov in x and y direction are used. Why is there a different parameter for them?

Reply:
to allow more flexibility for example in a case of filtering in the map frame. This does not restrict us in any way.